packages/relay/src/TransactionManager.ts

```diff
@@ -132,12 +134,13 @@ data                     | ${transaction.data}
     }
   }
 
-  async broadcastTransaction (signedTx: string, verifiedTxId: string, nonce: number): Promise<SignedTransactionDetails> {
+  async broadcastTransaction (signedTx: string, verifiedTxId: string, nonce: number, signer: string): Promise<SignedTransactionDetails> {
     try {
       const transactionHash = await this.contractInteractor.broadcastTransaction(signedTx)
       if (transactionHash.toLowerCase() !== verifiedTxId.toLowerCase()) {
         throw new Error(`txhash mismatch: from receipt: ${transactionHash} from txstore:${verifiedTxId}`)
       }
+      this.emit('TransactionBroadcast')
```

Comment:
Is this the only reason the `TransactionManager` we have had for years now has to be an `EventEmitter`? 
A single event that has to be handled by a single class?

I don't care that much about this server's codebase as it needs a complete rewrite at this point, but still, I think introducing event-driven code like this is abuse of the concept.

Maybe it is better to accept a single optional `onTransactionListener` variable in the constructor?